### PR TITLE
Fluent config validation

### PR DESCRIPTION
**Pull request by sowa705**
API for complex fluent validators

### CODE REVIEW

**Comment by ppiwow-apay** [comment]
would like to see some examples in readme at least - if possible please describe in readme previous thing and add usage examples of this Options validation please

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/OptionsBuilderFluentValidationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Allegro.Extensions.Validators;
+
+/// <summary>
+/// Fluent validation extensions for options builder
+/// </summary>
+public static class OptionsBuilderFluentValidationExtensions
+{
+    /// <summary>
+    /// Adds a scoped IValidateOptions&lt;TOptions&gt; service with fluent validation.
```
Comment (darthkurak) [inline src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/OptionsBuilderFluentValidationExtensions.cs:12]: we add as singleton, and summary tells about scoped

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    /// Adds a new IOptions&lt;TOptions&gt; to the service collection.
+    /// </summary>
+    public static OptionsBuilder<TOptions> AddWithFluentValidation<TOptions, TValidator>(
+        this IServiceCollection services,
+        string? configurationSection)
+        where TOptions : class
+        where TValidator : class, IValidator<TOptions>
+    {
+        // Add the validator
+        services.AddSingleton<IValidator<TOptions>, TValidator>();
```
Comment (darthkurak) [inline src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs:37]: should be scoped?

File: src/Allegro.Extensions.Validators/README.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+Registering all validators in assembly
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.RegisterFluentValidators(typeof(Startup).Assembly);
+}
+```
+
+Create IOptions<T> with fluent validation enabled
```
Comment (darthkurak) [inline src/Allegro.Extensions.Validators/README.md:20]: missing example of ValidateFluentValidation

**Review by darthkurak — CHANGES_REQUESTED** [review]

**Review by darthkurak — APPROVED** [review]

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/FluentValidationOptions.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Validators are typically registered as scoped,
+        // so we need to create a scope to be safe, as this
+        // method is be called from the root scope
+        using var scope = _serviceProvider.CreateScope();
+
+        // retrieve an instance of the validator
+        var validator = scope.ServiceProvider.GetRequiredService<IValidator<TOptions>>();
+
+        // Run the validation
```
Comment (orgiele) [inline src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/FluentValidationOptions.cs:50]: część tych komentarzy IMHO niewiele wnosi

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/FluentValidationOptions.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
+using System;
```
Comment (orgiele) [inline src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/FluentValidationOptions.cs:1]: jeżeli to jest kod skopiowany z jakiejś strony, to imho wypadałoby to zaznaczyć w jakimś komentarzu

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        return services.Scan(s => s.FromAssemblies(assemblies)
+            .AddClasses(c => c.AssignableTo(typeof(IValidator<>)))
+            .AsImplementedInterfaces()
+            .WithScopedLifetime());
+    }
+
+    /// <summary>
+    /// Adds a new IOptions&lt;TOptions&gt; to the service collection.
+    /// </summary>
+    public static OptionsBuilder<TOptions> AddWithFluentValidation<TOptions, TValidator>(
```
Comment (orgiele) [inline src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs:30]: AddWith brzmi dziwnie, albo Add albo With 😁

**Review by orgiele — CHANGES_REQUESTED** [review]

File: src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs (hunk 6 above)
Comment (sowa705) [reply src/Allegro.Extensions.Validators/Allegro.Extensions.Validators/StartupExtensions.cs:30]: może AddOptionsWithFluentValidation?

**Pull request merged**